ksql-rest-app/src/main/java/io/confluent/ksql/api/server/ProxyHandler.java

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright 2020 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.api.server;
+
+import io.vertx.core.Handler;
+import io.vertx.core.MultiMap;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.http.HttpClient;
+import io.vertx.core.http.HttpClientRequest;
+import io.vertx.core.http.HttpClientResponse;
+import io.vertx.core.http.HttpServerRequest;
+import io.vertx.core.http.HttpServerResponse;
+import io.vertx.core.net.SocketAddress;
+import io.vertx.core.streams.ReadStream;
+import io.vertx.core.streams.WriteStream;
+import io.vertx.ext.web.RoutingContext;
+import java.util.Map;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+public class ProxyHandler implements Handler<RoutingContext> {
+
+  private static final Logger log = LoggerFactory.getLogger(ProxyHandler.class);
+
+  private final HttpClient proxyClient;
+  private final Server server;
+  private SocketAddress proxyTarget;
+
+  public ProxyHandler(final SocketAddress proxyTarget, final HttpClient proxyClient,
+      final Server server) {
+    this.proxyTarget = proxyTarget;
+    this.proxyClient = proxyClient;
+    this.server = server;
+  }
+
+  @Override
+  public void handle(final RoutingContext routingContext) {
+    if (proxyTarget == null) {
+      proxyTarget = server.getProxyTarget();
+    }
+    final HttpServerRequest serverRequest = routingContext.request();
+    final HttpClientRequest clientRequest = proxyClient.request(serverRequest.method(),
+        proxyTarget, proxyTarget.port(), proxyTarget.host(),
+        serverRequest.path(),
+        resp -> responseHandler(resp, serverRequest))
+        .exceptionHandler(ProxyHandler::exceptionHandler);
+
+    final MultiMap headers = MultiMap.caseInsensitiveMultiMap();
+    for (Map.Entry<String, String> header : serverRequest.headers()) {
+      if (!header.getKey().equalsIgnoreCase("host")) {
```

Comment:
Out of curiosity, what goes wrong if the "host" header is not removed from the proxy request?

Reply:
It's probably not necessary, will try and remove